pkg/comp-functions/functions/helper/funcio.go

```
@@ -0,0 +1,236 @@
+	unstructured.RemoveNestedField(obj, "spec", "forProvider", "manifest", "metadata", "managedFields")
+
+	res.Raw, err = json.Marshal(&obj)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func createOrUpdateObject(ctx context.Context, configMap *corev1.ConfigMap, kubeClient client.Client) error {
+
+	tmpMap := &corev1.ConfigMap{}
+
+	err := kubeClient.Get(ctx, client.ObjectKeyFromObject(configMap), tmpMap)
+	if err != nil && apierrors.IsNotFound(err) {
+		err := kubeClient.Create(ctx, configMap)
+		if err != nil {
+			return err
+		}
+		return nil
+	} else if err != nil {
+		return nil
+	}
+
+	err = kubeClient.Update(ctx, configMap)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func diffFuncIOs(ctx context.Context, currentIndex int64, currentMap *corev1.ConfigMap, kubeClient client.Client, funcName string) error {
+
+	prevIndex := 0
+	if currentIndex == 0 {
+		prevIndex = 1
+	}
+
+	prevMap := &corev1.ConfigMap{}
+
+	prevMapName := strings.TrimRight(currentMap.GetName(), fmt.Sprintf("%d", currentIndex)) + fmt.Sprintf("%d", prevIndex)
+
+	prevMap.ObjectMeta.Name = prevMapName
+	prevMap.ObjectMeta.Namespace = "default"
+
+	err := kubeClient.Get(ctx, client.ObjectKeyFromObject(prevMap), prevMap)
+	if err != nil && apierrors.IsNotFound(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	prevPretty := bytes.Buffer{}
+	err = json.Indent(&prevPretty, []byte(prevMap.Data["funcIO"]), "", "  ")
+	if err != nil {
+		return err
+	}
+
+	currentPretty := bytes.Buffer{}
+	err = json.Indent(&currentPretty, []byte(currentMap.Data["funcIO"]), "", "  ")
+	if err != nil {
+		return err
+	}
+
+	diff := difflib.UnifiedDiff{
+		A:        difflib.SplitLines(prevPretty.String()),
+		B:        difflib.SplitLines(currentPretty.String()),
+		FromFile: "Previous",
+		ToFile:   "Current",
+		Context:  3,
+	}
+	text, _ := difflib.GetUnifiedDiffString(diff)
```

Nitpick: You shouldn't ignore the error, even if you don't ever expect one